crates/agent/src/publications.rs

```diff
@@ -328,6 +328,8 @@ impl PublishHandler {
             return stop_with_errors(errors, JobStatus::PublishFailed, row, txn).await;
         }
 
+        // ensure that this tempdir doesn't get dropped before `deploy_build` is called, which depends on the files being there.
+        std::mem::drop(tmpdir_handle);
```

Comment:
I'm actually not certain exactly why this ever worked in the first place. The `TempDir` has all contents deleted synchronously when it's dropped, so this _should_ have been failing consistently, since `tmpdir` should have been dropped as soon as the binding was overwritten (literally, the next line). I'm also not 100% clear on when the first `tmpdir` should be dropped, especially given the generated state machine for async functions. But I _can_ confirm that this change did fix the `no such file` errors I was getting during publish operations locally :man_shrugging: